### PR TITLE
Add dep constraints; use dep-server to get dep updates

### LINKED ISSUE #280: Consume the Go dependency from the Paketo dep server
Per https://github.com/paketo-buildpacks/dep-server/issues/45, we should start consuming the Go runtime dependency from the dep-server (See https://api.deps.paketo.io/v1/dependency?name=go for example dependency URIs)  instead of the https://buildpacks.cloudfoundry.org/dependencies/go/... location we currently get the dependencies from.

This will make the dependency publishing/consumption process more transparent than the process we use for the dependencies available via the [dependency-builds pipeline](https://buildpacks.ci.cf-app.com/teams/main/pipelines/dependency-builds).

We have already done this switch-over in the Node Engine and Yarn Buildpacks. The outline of what this work will entail can be found in the dep-server issue linked at the top.

### PR DESCRIPTION
**Pull request by fg-j**
## Summary

Resolves #280 

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [ ] I have added an integration test, if necessary.
* [ ] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [ ] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: buildpack.toml [hunk 1]
```diff
@@ -62,6 +62,21 @@ api = "0.5"
     uri = "https://buildpacks.cloudfoundry.org/dependencies/go/go_1.17_linux_x64_cflinuxfs3_668718ba.tgz"
     version = "1.17"
 
+  [[metadata.dependency-constraints]]
+    constraint = "1.15.*"
+    id = "go"
+    patches = 2
```
Comment (ForestEckhardt) [inline buildpack.toml:68]: Should we not add this constraint and remove the 1.15.x line as 1.17.x is out?

Comment (fg-j) [reply buildpack.toml:68]: Good question. I'm not sure. I only translated what was currently in the `buildpack.toml` in terms of versions into the dependency constraint format. If we remove 1.15, do we need a EOL notice // should we announce somehow?

Comment (fg-j) [reply buildpack.toml:68]: I'm going to merge as-is for now. We can decide re: 1.15 later.

Comment (ForestEckhardt) [reply buildpack.toml:68]: Sounds good to me. Will review.

**Comment by fg-j** [comment]
Concourse pipelines for updating this dependency have been turned off: https://github.com/cloudfoundry/buildpacks-ci/commit/905c8e4b9916fa27736a0db5dc8450f66df60a0d

**Review by ForestEckhardt — APPROVED** [review]

**Pull request merged**